assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentToolUtils.java

```diff
@@ -265,552 +158,69 @@ public List<String> alertInvalidPoint(String grade, int factor) {
     }
 
     /**
-     * Common grading routine plus specific operation to differentiate cases when saving, releasing or returning grade.
+     * scale the point value by "factor" if there is a valid point grade
      */
-    public void gradeSubmission(AssignmentSubmission submission, String gradeOption, Map<String, Object> options, List<String> alerts) {
-
-        if (submission != null) {
-            boolean withGrade = options.get(WITH_GRADES) != null && (Boolean) options.get(WITH_GRADES);
-            String grade = (String) options.get(GRADE_SUBMISSION_GRADE);
-            boolean gradeChanged = false;
-            if (!StringUtils.equals(StringUtils.trimToNull(submission.getGrade()), StringUtils.trimToNull(grade))) {
-                //one is null the other isn't
-                gradeChanged = true;
-            }
-            Assignment a = submission.getAssignment();
-            if (!withGrade) {
-                // no grade input needed for the without-grade version of assignment tool
-                submission.setGraded(true);
-                if (gradeChanged) {
-                    submission.setGradedBy(userDirectoryService.getCurrentUser() == null ? null : userDirectoryService.getCurrentUser().getId());
-                }
-                if (SUBMISSION_OPTION_RETURN.equals(gradeOption) || SUBMISSION_OPTION_RELEASE.equals(gradeOption)) {
-                    submission.setGradeReleased(true);
-                }
-            } else {
-                //If the grade is not blank
-                if (StringUtils.isNotBlank(grade)) {
-                    submission.setGrade(grade);
-                    submission.setGraded(true);
-                    if (gradeChanged) {
-                        submission.setGradedBy(userDirectoryService.getCurrentUser() == null ? null : userDirectoryService.getCurrentUser().getId());
-                    }
-                } else {
-                    submission.setGrade(null);
-                    submission.setGraded(false);
-                    if (gradeChanged) {
-                        submission.setGradedBy(null);
-                    }
-                }
-            }
+    public static String scalePointGrade(String point, int factor, List<String> alerts) {
```

Comment:
this should be removed use the method in the service

Reply:
I don't want to do this in this PR. I don't want to break anything.